index.js

```diff
@@ -31,6 +33,13 @@ const preprocessDestinationPath = (source, destination, options) => {
 
 module.exports = (source, destination, options = {}) => {
 	const progressEmitter = new EventEmitter();
+	let concurrencyDegree;
```

Comment:
Why did you create a new var to make `options.concurrency` default to `(os.cpus().length || 1) * 2` ?

Reply:
I added the new variable to make it explicit in reading what the value meant, do you think it's unnecessary? I could make it so no variable is added, my only concern was the readability of the code.

Reply:
I think what @hugomrdias is saying (and giving you a hint) is that you can simply the code. For example, you may leverage object spread to have default options being mixed in with the ones the user supplies.

Reply:
Hmm, not sure what you mean with object spread, since this code change operates with only one of the options. The other options are handled in other functions.

I could default it like:
`module.exports = (source, destination, options = {concurrency: (os.cpus().length || 1 * 2}) => [...]`
...which also makes the default value easy to find. Do you think that would be better?

Reply:
You are right in the sense that each option is currently being handled in each function, so perhaps lets keep the consistency.

Still, we may simplify with:

```js
    const concurrency = options.concurrency || (os.cpus().length || 1) * 2;
```

...and use it directly. Also, by renaming the variable to `concurrency` from `concurrencyDegree`, you may use it directly when passing it to `pAll`.

Other than that, the PR looks good. However, I would add a simple test to see if the concurrency is being respected.